src/StateService/Verification/VerificationContext.cs

```diff
@@ -111,17 +120,20 @@ public bool CheckSignatures()
         {
             if (StateRoot is null) return false;
             if (signatures.Count < M) return false;
-            if (StateRoot.Witness != null) return true;
-            Contract contract = Contract.CreateMultiSigContract(M, verifiers);
-            ContractParametersContext sc = new ContractParametersContext(StatePlugin.System.StoreView, StateRoot);
-            for (int i = 0, j = 0; i < verifiers.Length && j < M; i++)
+            if (StateRoot.Witness is null)
             {
-                if (!signatures.TryGetValue(i, out byte[] sig)) continue;
-                sc.AddSignature(contract, verifiers[i], sig);
-                j++;
+                Contract contract = Contract.CreateMultiSigContract(M, verifiers);
+                ContractParametersContext sc = new ContractParametersContext(StatePlugin.System.StoreView, StateRoot);
+                for (int i = 0, j = 0; i < verifiers.Length && j < M; i++)
+                {
+                    if (!signatures.TryGetValue(i, out byte[] sig)) continue;
```

Comment:
```suggestion
                    if (!signatures.TryGetValue(i, out byte[] sig)) return false;
```

Reply:
Why? It is uncertain from which verifiers the signature was received.